src/device_constructors/renewablegeneration_constructor.jl

```diff
@@ -0,0 +1,95 @@
+function construct_device!(ps_m::CanonicalModel,
+                           device::Type{R},
+                           device_formulation::Type{D},
+                           system_formulation::Type{S},
+                           sys::PSY.PowerSystem;
+                           kwargs...) where {R <: PSY.RenewableGen,
+                                             D <: AbstractRenewableDispatchForm,
+                                             S <: PM.AbstractPowerFormulation}
+
+    #Defining this outside in order to enable time slicing later
+    time_range = 1:sys.time_periods
+
+    fixed_resources = [fs for fs in sys.generators.renewable if isa(fs,PSY.RenewableFix)]
```

Comment:
This code is iterating over the same array twice.  It would be more clear (and do less work) if there was one for loop and then an if/else clause for the type of sys.generators.renewable.  Secondly, is there a way to specify an exact type in the second case (instead of !isa RenewableFix)?

Reply:
Noted, this needs to be resolved with a better classification code. What we do now is get the elements of the renewable energy Array that are fixed we have a bunch of other types that aren't fixed type. 

I added a note to be resolved in the next iteration.